channeldb/kvdb/etcd/bucket.go

```diff
@@ -0,0 +1,86 @@
+package etcd
+
+import (
+	"crypto/sha256"
+)
+
+const (
+	bucketIDLength = 32
+)
+
+var (
+	bucketPrefix   = []byte("b")
+	valuePrefix    = []byte("v")
+	sequencePrefix = []byte("$")
+)
+
+// rootBucketId returns a zero filled 32 byte array
+func rootBucketID() []byte {
+	var rootID [bucketIDLength]byte
+	return rootID[:]
+}
+
+// makeBucketID returns a deterministic key for the passed byte slice.
+// Currently it returns the sha256 hash of the slice.
+func makeBucketID(key []byte) [bucketIDLength]byte {
+	return sha256.Sum256(key)
```

Comment:
An alternative to creating the bucket ID based on the actual bucket _name_ would be to use a sequence number here. So the first bucket has an ID of zero, the next one and so one. I don't envision in the short term that we'll run into performance issues with usage of `sha2` here, but if we want to drop tat dependnacy, then we can go with the sequence number route instead.

Reply:
This would not work if there are multiple db instances. Also we need to make sure bucket ids are deterministic when running multiple nodes on the same db (in the far-far future :))

Reply:
Also the sequence number thing would require deterministic order of bucket creation, otherwise failover would be tricky.

Reply:
Ah makes sense, great decision.